### ISSUE TITLE
Can't lock directories exclusively

### ISSUE THREAD

**Issue by Kritzefitz** [hesselink/flock#11]
I tried to lock a directory exclusively using `lock "/home" Exclusive Block`. Unfortunately it didn't work an failed with this exception:

```
*** Exception: /home: openFd: inappropriate type (Is a directory)
```

This happens because `openFd` can open directories only `ReadOnly`. Opening directories with `ReadOnly` is sufficient for locking them exclusively with `flock` as shown with this code:

```
lockDirectoryExample = do
  fd <- openFd "/home" ReadOnly Nothing defaultFileFlags
  l <- lockFd fd
```

 I guess to fix this `lock` would have to check if the given path is a directory and if so, open it with `ReadOnly`.

**Comment by hesselink** [comment]
I've looked at this briefly, and I can't find if it's necessary to open a normal file in write mode for an exclusive lock at all. So maybe we can get away with just opening the file or directory in read mode every time.

**Comment by Kritzefitz** [comment]
Ah, yes I just tried locking an FD opened with `ReadOnly` at it seems to work. I couldn't find any reliable source for the need for a write lock either. Apparently I just got confused by this wrong [stackoverflow answer](https://stackoverflow.com/a/8817125/2211273)

**Comment by hesselink** [comment]
This completely slipped my mind! I just fixed it, and released as 0.3.2
